lib/core-integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -1045,6 +1078,30 @@ getFromResponseList i getter (_, res) = case res of
 json :: QuasiQuoter
 json = aesonQQ
 
+joinStakePool
+    :: forall t w. (HasType (ApiT WalletId) w)
+    => Context t
+    -> ApiStakePool
+    -> (w, Text)
+    -> IO (HTTP.Status, Either RequestException [ApiStakePool])
+joinStakePool ctx p (w, pass) = do
+    let payload = Json [aesonQQ| {
+            "passphrase": #{pass}
+            } |]
+    request @[ApiStakePool] ctx (joinStakePoolEp p w) Default payload
+
+quitStakePool
+    :: forall t w. (HasType (ApiT WalletId) w)
+    => Context t
+    -> ApiStakePool
+    -> (w, Text)
+    -> IO (HTTP.Status, Either RequestException [ApiStakePool])
+quitStakePool ctx p (w, pass) = do
+    let payload = Json [aesonQQ| {
+            "passphrase": #{pass}
+            } |]
+    request @[ApiStakePool] ctx (quitStakePoolEp p w) Default payload
```

Comment:
Similarly, the return type here should be `ApiTransaction` rather than `[ApiStakePool]`.